### PR TITLE
Nss complaint

### PR DESCRIPTION
**Pull request by sshivam95**

### CODE REVIEW

File: .idea/IITBapp.iml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    <option name="TEMPLATE_FOLDERS">
+      <list>
+        <option value="$MODULE_DIR$/prerender/templates" />
+      </list>
+    </option>
+  </component>
+  <component name="TestRunnerService">
+    <option name="PROJECT_TEST_RUNNER" value="Unittests" />
+  </component>
+</module>
```
Comment (pulsejet) [inline .idea/IITBapp.iml]: Don't commit IDE specific folders (.idea)

File: backend/settings_base.py [hunk 2]
```diff
@@ -35,7 +35,7 @@
     'news.apps.NewsConfig',
     'messmenu.apps.MessmenuConfig',
     'other.apps.OtherConfig',
-
+    'venter',
```
Comment (pulsejet) [inline backend/settings_base.py]: `venter.apps.VenterConfig` instead of just `venter`. This is the new convention.

File: backend/urls.py [hunk 3]
```diff
@@ -14,9 +14,10 @@
     2. Add a URL to urlpatterns:  path('blog/', include('blog.urls'))
 """
 from django.contrib import admin
-from django.urls import path
+from django.urls import path, include
 from rest_framework_swagger.views import get_swagger_view
 
+import venter
```
Comment (pulsejet) [inline backend/urls.py]: Why?

File: backend/urls.py [hunk 4]
```diff
@@ -34,74 +35,77 @@
 urlpatterns = [
     path('admin/', admin.site.urls),
 
-    path('api/bodies', BodyViewSet.as_view({'get':'list', 'post':'create'})),
+    path('api/bodies', BodyViewSet.as_view({'get': 'list', 'post': 'create'})),
     path('api/bodies/<pk>', BodyViewSet.as_view(
```
Comment (pulsejet) [inline backend/urls.py:38]: No unnecessary whitespace changes please; make a separate PR to master for this if necessary.

File: backend/urls.py [hunk 5]
```diff
@@ -34,74 +35,77 @@
...
 
-    path('api/search', OtherViewset.as_view({'get':'search'})),
-    path('api/notifications', OtherViewset.as_view({'get':'get_notifications'})),
-    path('api/notifications/read', OtherViewset.as_view({'get':'mark_all_notifications_read'})),
-    path('api/notifications/read/<pk>', OtherViewset.as_view({'get':'mark_notification_read'})),
+    path('api/search', OtherViewset.as_view({'get': 'search'})),
+    path('api/notifications', OtherViewset.as_view({'get': 'get_notifications'})),
+    path('api/notifications/read', OtherViewset.as_view({'get': 'mark_all_notifications_read'})),
+    path('api/notifications/read/<pk>', OtherViewset.as_view({'get': 'mark_notification_read'})),
+    ##
```
Comment (pulsejet) [inline backend/urls.py]: Remove the `##`

File: bodies/admin.py [hunk 6]
```diff
@@ -1,5 +1,15 @@
...
+class RoleInline(admin.StackedInline):
+    model = BodyRole
+    fields = ('name',)
+    readonly_fields = ('name',)
+    show_change_link = True
+
+class BodyAdmin(admin.ModelAdmin):
+    inlines = [RoleInline]
+
+admin.site.register(Body, BodyAdmin)
```
Comment (pulsejet) [inline bodies/admin.py]: Why do we need these changes?

File: venter/models.py [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+
+    class Meta:
+        verbose_name = "Comment"
+        verbose_name_plural = "Comments"
+
+    def __str__(self):
+        return self.text
+
+
+        # class complaints_liked_users(models.Model):
```
Comment (pulsejet) [inline venter/models.py]: No commented code please. See [this](https://softwareengineering.stackexchange.com/questions/190096/can-commented-out-code-be-valuable-documentation) thread.

File: venter/serializers.py [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+
+
+class ComplaintPostSerializer(serializers.ModelSerializer):
+    tags = TagSerializer(many=True, read_only=True)
+    tag_names = serializers.SlugRelatedField(many=True, read_only=True, slug_field='tag_uri', source='tags')
+    tag_ids = serializers.PrimaryKeyRelatedField(many=True, read_only=False, source='tags',
+                                                 queryset=tag_uris.objects.all(), required=True)
+
+    media = UploadedImageSerializer(many=True, read_only=True)
+    # media = serializers.SlugRelatedField(many=True, read_only=True, slug_field='picture', source='media')
```
Comment (pulsejet) [inline venter/serializers.py]: No commented code, again

File: venter/views.py [hunk 9]
```diff
@@ -0,0 +1,28 @@
+from coreapi.exceptions import ParseError
+from django.shortcuts import render
+
+# Create your views here.
+from rest_framework.decorators import detail_route, action
+
+from .models import complaints
```
Comment (pulsejet) [inline venter/views.py]: Use `venter.models` and similar instead of just `.models`

**Review by pulsejet — CHANGES_REQUESTED** [review]
A few notes:
1) You'll need to add test cases for all APIs
2) Merge all migrations once everything looks right
3) Some general issues reported by Codacy should be fixed.
I don't have time right now to test this; will do it as soon as possible.

**Comment by sshivam95** [comment]
Noted.

FYI I haven't changed anything in the bodies/admin.py, it may be because of
the previous commit when i cloned the repo.

Regards
Shivam Sharma

**Comment by pulsejet** [comment]
Yes, those changes are probably from b23891d59768f2990f10fe66e2a67f700300f89b. Just revert them.

**Review by pulsejet — APPROVED** [review]

**Comment by pulsejet** [comment]
Closing this in favour of #211

**Pull request closed without merging**